src/github.com/getlantern/yamlconf/yamlconf_disk.go

```diff
@@ -25,13 +31,30 @@ func (m *Manager) reloadFromDisk() (bool, error) {
 		log.Trace("Config unchanged on disk")
 		return false, nil
 	}
-	bytes, err := ioutil.ReadFile(m.FilePath)
-	if err != nil {
-		return false, fmt.Errorf("Error reading config from %s: %s", m.FilePath, err)
-	}
-	err = yaml.Unmarshal(bytes, cfg)
-	if err != nil {
-		return false, fmt.Errorf("Error unmarshaling config yaml from %s: %s", m.FilePath, err)
+
+	var cfg Config
+	if m.Obfuscate {
+		log.Trace("Attempting to read obfuscated config")
+		var err1, err2 error
+		cfg, err1 = m.doReadFromDisk(true)
+		if err1 != nil {
+			log.Tracef("Error reading obfuscated config from disk, try reading non-obfuscated: %v", err)
+			cfg, err2 = m.doReadFromDisk(false)
+			if err2 != nil {
+				return false, err1
```

Comment:
In this case, why are we more interested in the first error? Perhaps return both concatenated?